software/firmware/src/board/bluepill.rs

```diff
@@ -1,16 +1,30 @@
 use embedded_hal::PwmPin;
 use pac::TIM2;
+use stm32f1xx_hal::gpio::gpioa::{PA10, PA8, PA9};
+use stm32f1xx_hal::gpio::gpiob::{PB13, PB14, PB15};
+use stm32f1xx_hal::gpio::{Alternate, Floating, IOPinSpeed, Input, Output, OutputSpeed, PushPull};
+use stm32f1xx_hal::pac::SPI2;
```

Comment:
Style: Line 10 gives the pac crate a short name. You can either use that name here, or remove line 10 and rewrite this as `stm32f1xx_hal::pac::{SPI2, TIM2}`.